tree/treeplayer/inc/TMPWorkerTree.h

```diff
@@ -35,8 +35,9 @@ class TMPWorkerTree : public TMPWorker {
    /// \endcond
 public:
    TMPWorkerTree();
-   TMPWorkerTree(const std::vector<std::string>& fileNames, const std::string& treeName, unsigned nWorkers, ULong64_t maxEntries);
-   TMPWorkerTree(TTree *tree, unsigned nWorkers, ULong64_t maxEntries);
+   TMPWorkerTree(const std::vector<std::string> &fileNames, TEntryList *entries, const std::string &treeName,
```

Comment:
> All codes use the fact that 'entries == nullptr' 

This is in the inner function.  For interface purpose we could have a user level interface that takes a reference and a inner/private functions that is shared among implementation with TEntryList being a nullptr.